### PR TITLE
6 Reasons Why I get Annoyed Easily

### PR DESCRIPTION
**Pull request by ribagi**
1: String Concatenation is Bad
2: If null object, return null. If not null object return object is a
weird statement.
3: Constants should be constant.
4: While you can initialize outside of the constructor, it is still
kinda weird imo.
5:
6: There are some if statements that do nothing.

Civchat just had so many problems that I had to fix them.

### CODE REVIEW

**Comment by ProgrammerDan** [comment]
+1 for fixing what bugs ya. String concatenation doesn't bug me like it does you, but it's nice to see the proper pattern applied in any case. Don't look too closely at my latest commits on Humbug, you might get annoyed (although that whole mod needs a SERIOUS review anyway, same as all the others). Good on ya!

File: src/vg/civcraft/mc/civchat2/CivChat2Manager.java [hunk 1]
```diff
@@ -142,9 +154,22 @@ public void groupChat(String chatGroupName, String chatMessage, String msgSender
 				continue;
 			}
 			else{
-				String grayMessage = ChatColor.GRAY + "[" + groupName + "] " + msgSender + ": ";
-				String whiteMessage = ChatColor.WHITE + chatMessage;
-				p.sendMessage(grayMessage + whiteMessage);
+				String grayMessage = sb.append(ChatColor.GRAY)
+										.append( "[")
```
Comment (ProgrammerDan) [inline src/vg/civcraft/mc/civchat2/CivChat2Manager.java:158]: Just my old eyes, but I suppose this is why I don't mind concatenation. Having a bunch of danglies off in the right margin somewhere makes my head hurt.

Also, [your rage is unfounded](http://stackoverflow.com/questions/1532461/stringbuilder-vs-string-concatenation-in-tostring-in-java) -- the JVM is likely just converting raw concats into the StringBuilder pattern behind the scenes, stop letting it anger you and embrace the enhanced readability.

**Comment by ribagi** [comment]
In the jvm each time a string gets concated, it creates a new instance of stringbuilder. So while I am fine if within a single scope only one concate happens,  if any more than one happens then there are just waste memory for nothing really good. This is boosted if the string concate happens in a loop, which happened a few times in this code.

**Comment by ProgrammerDan** [comment]
I'm aware. To be honest didn't look that closely at the specifics here; naturally if concatenation is happening in the context of a loop, kill it with fire. Otherwise, you're wasting effort and anger-time.

Edit: Also the rule isn't quite that simple. Successive concatenations on the same line, for instance, do not each generate a new instance of StringBuilder iirc. However, concatenations that span loops or logic will definitely create new instances, which is wasteful.

**Comment by ttk2** [comment]
So should this get merged?

**Comment by ProgrammerDan** [comment]
Stylistic differences aside, i see no regressions, should be safe to merge.

**Comment by ttk2** [comment]
so should this go on Civtest?

**Pull request merged**